### PR TITLE
Base.v.16.0 doesn't build on windows

### PR DESCRIPTION
**Pull request by hhugo**
Here is the corresponding GH issue on the base repo https://github.com/janestreet/base/issues/150

### CODE REVIEW

**Comment by hhugo** [comment]
cc @dra27

**Comment by mseri** [comment]
For reference
```
#=== ERROR while compiling base.v0.16.0 =======================================#
# context     2.0.10 | win32/x86_64 | ocaml-variants.4.14.1+mingw64c | git+https://github.com/ocaml/opam-repository.git
# path        D:/a/js_of_ocaml/js_of_ocaml/_opam/.opam-switch/build/base.v0.16.0
# command     D:\a\js_of_ocaml\js_of_ocaml\_opam\bin\dune.exe build -p base -j 2
# exit-code   1
# env-file    D:/.opam/log/base-1484-ccda7f.env
# output-file D:/.opam/log/base-1484-ccda7f.out
### output ###
# File "src/dune", line 10, characters 0-500:
# 10 | (library (name base) (public_name base)
# 11 |  (ocamlopt_flags :standard (:include ocamlopt-flags))
# 12 |  (libraries base_internalhash_types caml sexplib0 shadow_stdlib)
# 13 |  (c_flags :standard -D_LARGEFILE64_SOURCE (:include mpopcnt.sexp))
# 14 |  (c_names bytes_stubs exn_stubs int_math_stubs hash_stubs am_testing)
# 15 |  (preprocess no_preprocessing)
# 16 |  (lint
# 17 |   (pps ppx_base ppx_base_lint -check-doc-comments -type-conv-keep-w32=both
# 18 |    -apply=js_style,base_lint,type_conv,cold))
# 19 |  (js_of_ocaml (javascript_files runtime.js)))
# (cd _build/default && D:\a\js_of_ocaml\js_of_ocaml\_opam\bin\ocamlopt.opt.exe -w -40 -g -shared -linkall -I src -o src/base.cmxs src/base.cmxa)
# /usr/lib/gcc/x86_64-w64-mingw32/11/../../../../x86_64-w64-mingw32/bin/ld: D:\cygwin\tmp\build_8402e2_dune\dyndll413622.o::(.data+0x3378): undefined reference to `caml_csel_value'
# /usr/lib/gcc/x86_64-w64-mingw32/11/../../../../x86_64-w64-mingw32/bin/ld: D:\cygwin\tmp\build_8402e2_dune\dyndllc7ad59.o::(.data+0x2e60): undefined reference to `caml_csel_value'
# /usr/lib/gcc/x86_64-w64-mingw32/11/../../../../x86_64-w64-mingw32/bin/ld: D:\cygwin\tmp\build_8402e2_dune\dyndll933c30.o::(.data+0xe60): undefined reference to `caml_csel_value'
# /usr/lib/gcc/x86_64-w64-mingw32/11/../../../../x86_64-w64-mingw32/bin/ld: D:\cygwin\tmp\build_8402e2_dune\dyndlle4b807.o::(.data+0xc0): undefined reference to `caml_csel_value'
# /usr/lib/gcc/x86_64-w64-mingw32/11/../../../../x86_64-w64-mingw32/bin/ld: D:\cygwin\tmp\build_8402e2_dune\dyndll6c0139.o::(.data+0x1b90): undefined reference to `caml_csel_value'
# /usr/lib/gcc/x86_64-w64-mingw32/11/../../../../x86_64-w64-mingw32/bin/ld: D:\cygwin\tmp\build_8402e2_dune\dyndll5105e7.o::(.data+0x1b50): more undefined references to `caml_csel_value' follow
# collect2: error: ld returned 1 exit status
# ** Fatal error: Error during linking
# 
# File "caml_startup", line 1:
# Error: Error during linking (exit code 2)
```

**Comment by mseri** [comment]
I'll wait for @dra27 before doing anything

**Review by dra27 — APPROVED** [review]
Sorry for the slowness - LGTM!

File: packages/base/base.v0.16.0/opam [hunk 1]
```diff
@@ -15,7 +15,7 @@ depends: [
   "dune"              {>= "2.0.0"}
   "dune-configurator"
 ]
-available: arch != "arm32" & arch != "x86_32"
+available: arch != "arm32" & arch != "x86_32" & os != "win32"
```
[the remark this answers is not in the document]
Comment (dra27) [reply packages/base/base.v0.16.0/opam]: `os-family = "windows"` is often likely to be a mistake, `os != "windows"` is an actual mistake (see #23946)

**Pull request merged**